test/e2e/util.go

```diff
@@ -81,6 +81,24 @@ func waitForPodCondition(c *client.Client, ns, podName, desc string, condition p
 	return fmt.Errorf("gave up waiting for pod %s to be %s after %.2f seconds", podName, desc, podStartTimeout.Seconds())
 }
 
+//monotonically increasing ns index is useful for certain
+//testing scenarios, but dont worry, we won't rely on it for a unique ID.
+var ginkgoNsIndex = 0
+
+// createNS should be used by every test, note that we append a common prefix to the provided test name.
+func createGinkgoNS(baseName string, c *client.Client) (*api.Namespace, error) {
+	ginkgoNsIndex++
+	namespaceObj := &api.Namespace{
+		ObjectMeta: api.ObjectMeta{
+			Name:      fmt.Sprintf("e2e-tests-%v-%v-%v", baseName, ginkgoNsIndex, strconv.FormatInt(time.Now().Unix(), 10)),
```

Comment:
Ok to have uuid with timestamp?  That way there is useful information with the ns. Recall name spaces are quite fragile and will be some time before we can really rely on them to life cycle flawlessly.. so more info is better